### DISCUSSION TITLE
Mono audio conversion.

### DISCUSSION THREAD

**Discussion by maxi1134** [music-assistant/support#3281] [Feature requests and ideas]
I currently use up to 12 speakers at once, in 6 different rooms.

When doing so I like to set those speakers to "mono" within their wiim configuration. (To avoid sound weirdnesses when pacing around)

This is cumbersome as I need to revert to Stereo before using the Wiim for one-room audio.

A function where we can set a Group, or a speaker, to "mono" would be great, although I don't know how hard it would be to remux the live stream to mono from stereo.

**Comment by OzGav** [comment]
You can already set AirPlay devices to mono

**Reply by maxi1134** [reply]
> You can already set AirPlay devices to mono

Where exactly can i do the switch? And can it be done with an action call?

**Reply by OzGav** [reply]
In the player settings. Let me look for an API call

**Reply by OzGav** [reply]
Get the player_id from the top of the player settings. I haven’t tested this so it may not work. You will have to enable port 8095 in the web server settings as well. To revert,  create another command with the value set to stereo

```
rest_command:  
  ma_set_player_output_channels_mono:
    url: http://localhost:8095/api
    method: POST
    headers:
      accept: "application/json, text/html"
    payload: >
      {
        "message_id": "1",
        "command": "config/players/save",
        "args": {
          "player_id": "b8:27:eb:8a:b8:8e",
          "values": {
            "output_channels": “mono”
          }
        }
      }
    content_type:  'application/json; charset=utf-8'
    
```

**Reply by maxi1134** [reply]
> Get the player_id from the top of the player settings. I haven’t tested this so it may not work. You will have to enable port 8095 in the web server settings as well. To revert, create another command with the value set to stereo
> 
> ```
> rest_command:  
>   ma_set_player_output_channels_mono:
>     url: http://localhost:8095/api
>     method: POST
>     headers:
>       accept: "application/json, text/html"
>     payload: >
>       {
>         "message_id": "1",
>         "command": "config/players/save",
>         "args": {
>           "player_id": "b8:27:eb:8a:b8:8e",
>           "values": {
>             "output_channels": “mono”
>           }
>         }
>       }
>     content_type:  'application/json; charset=utf-8'
>     
> ```

Would it be possible to have this as a toggle under the device in HA?

**Reply by OzGav** [reply]
Probably. You would have to use an input Boolean and some automations

**Reply by maxi1134** [reply]
> Probably. You would have to use an input Boolean and some automations

I meant more as a default entity from the intergration.

Would be very usefull for people having multiple sets that switch between groups and individual playback.

**Reply by OzGav** [reply]
No the integration doesn’t work like that. It provides actions to interact with MA. The toggle (actually a drop down list) is in the MA settings. So I expect you would set the usual config in MA and use the rest command to adjust as necessary.
